### PR TITLE
Ct 1139 add station form

### PR DESCRIPTION
**Pull request by Krankenwagen**

### CODE REVIEW

File: src/features/stations/components/StationForm.js [hunk 1]
```diff
@@ -1,91 +1,363 @@
...
+                          <Button
+                            type='button'
+                            onClick={() =>
+                              values.tags.includes(createTag)
+                                ? Object.assign(errors, { tags: 'tag exists' })
+                                : (arrayHelpers.push(createTag),
+                                  setCreateTag(''))
+                            }
+                          >
+                            Add{' '}
```
Comment (aurimas-darguzis) [inline src/features/stations/components/StationForm.js]: whitespace?

File: src/features/stations/components/StationForm.js [hunk 2]
```diff
@@ -1,91 +1,363 @@
...
+              type='button'
+              variant='contained'
+              color='secondary'
+              onClick={() => cancelStation(values.id)}
+              className={classes.button}
+              disabled={isSubmitting}
+            >
+              Cancel
+            </Button>
+            <span style={{ float: 'right' }}>
```
Comment (aurimas-darguzis) [inline src/features/stations/components/StationForm.js]: haven't seen float in a while now 😄

File: src/features/stations/components/StationForm.js [hunk 3]
```diff
@@ -1,91 +1,363 @@
...
-                onClick={handleSubmit}
+                type='submit'
+                variant='contained'
+                color='secondary'
+                onClick={() => resetForm(initialValues)}
                 className={classes.button}
                 disabled={isSubmitting}
               >
-                Edit
+                <RotateLeftIcon />
```
Comment (aurimas-darguzis) [inline src/features/stations/components/StationForm.js]: goood stuff 😎

File: src/features/stations/components/StationForm.js (hunk 1 above)
Comment (Krankenwagen) [reply src/features/stations/components/StationForm.js]: formatter. i swer

File: src/features/stations/components/StationForm.js [hunk 4]
```diff
@@ -1,91 +1,363 @@
...
+            {parents.map(option => (
+              <MenuItem key={option.value} value={option.value}>
+                {option.label}
+              </MenuItem>
+            ))}
+          </TextField>
+        </div>
+        <div className={classes.formBlock}>
+          <div className={classes.formRow}>
+            <FieldArray
```
Comment (mohaq-global) [inline src/features/stations/components/StationForm.js]: Functionally this works but I was wondering if something like https://react-select.com/creatable (Creatable Multiselect example) will work better here as it tracks state internally and prevents duplicates from being typed. We are using this library for linked source multiselect

File: src/features/stations/components/StationForm.js [hunk 5]
```diff
@@ -1,91 +1,363 @@
...
+                    Add IP Bearer{' '}
+                  </Button>
+                </div>
+              )}
+            />
+          </div>
+          <div className={classes.formBlock}>
+            <div className={classes.formRow}>
+              {createTag}
+              <FieldArray
```
Comment (mohaq-global) [inline src/features/stations/components/StationForm.js]: same as above

File: src/features/stations/components/StationForm.js [hunk 6]
```diff
@@ -1,91 +1,363 @@
...
 
 const useStyles = makeStyles(theme => ({
   textField: {
     marginLeft: theme.spacing(1),
     marginRight: theme.spacing(1),
+    marginTop: theme.spacing(2),
     width: 200
   },
+  urlField: {
+    width: 400,
```
Comment (aurimas-darguzis) [inline src/features/stations/components/StationForm.js]: We can look into this in a separate ticket, but I wouldn't use fixed width. I think better approach is to use gird and give fractions or % and then justify the content accordingly. Then we are not tied to any specific screen size, while this approach can potentially create styling issue in the future. 🥔

File: src/features/stations/components/StationForm.js (hunk 4 above)
Comment (Krankenwagen) [reply src/features/stations/components/StationForm.js]: Ah, that would be a great switch then 👍

File: src/features/stations/components/StationForm.js [hunk 7]
```diff
@@ -1,91 +1,363 @@
...
+                    onClick={() =>
+                      arrayHelpers.push({
+                        url: 'http://media-ice.musicradio.com/',
+                        format: 'audio/aacp',
+                        bitrate: 48,
+                        cost: 30
+                      })
+                    }
+                  >
+                    Add IP Bearer{' '}
```
Comment (aurimas-darguzis) [inline src/features/stations/components/StationForm.js]: whitespace?

File: src/features/stations/components/StationForm.js (hunk 6 above)
Comment (Krankenwagen) [reply src/features/stations/components/StationForm.js]: Agreed, sounds like a more scalable option 🥇

File: src/features/stations/components/StationForm.js (hunk 1 above)
Comment (aurimas-darguzis) [reply src/features/stations/components/StationForm.js]: image

File: src/features/stations/components/StationForm.js (hunk 4 above)
Comment (Krankenwagen) [reply src/features/stations/components/StationForm.js]: Are you sure we are using this? Can't find it in linked sources 🤔 ☁️

Comment (mohaq-global) [reply src/features/stations/components/StationForm.js]: its in linkedSourcesForm.js 
only using the normal Select whereas here you'll need Creatable. Give it a shot to see if its worth it

Comment (Krankenwagen) [reply src/features/stations/components/StationForm.js]: @mohaq-global Sounds good, i'll give it a go (:

**Pull request merged**